### DISCUSSION TITLE
About the new bevy::render::FrameCountPlugin

### DISCUSSION THREAD

**Discussion by phuocthanhdo** [bevyengine/bevy#6656] [Q&A]
Hi, I was wondering why the FrameCount is only increment with a system in bevy::render crate?

It leads to the `FrameTimeDiagnosticsPlugin` cannot update frame_count without adding the FrameCountPlugin.

1. The app will need to pull a lot dependencies from `bevy::render` to add FrameCountPlugin.
2. Or manually fix `FrameTimeDiagnosticsPlugin ` by making another system like that if don't want to import `bevy::render` crate

**Comment by alice-i-cecile** [comment]
This was due to timing consistency challenges IIRC. Decision was made by @hymm. Perhaps we should reorganize where this system is stored, and then call it from bevy_render?

**Reply by hymm** [reply]
FYI I didn't move the FrameCountPlugin in my PR.

I think the organization is more that FrameCountPlugin has an optional dependency on RenderPlugin and the global uniform plugin depends on FrameCountPlugin.

**Reply by hymm** [reply]
It might make sense to split up FrameCountPlugin and move the update into bevy_core (where FrameCount is defined) or bevy_diagnostics and just leave the extract step for the FrameCount in `bevy_render` or move it to the globals.rs file.

https://github.com/bevyengine/bevy/blob/9f51651eacddc2267f88a3b1a44af1ef58ec5d73/crates/bevy_render/src/lib.rs#L372-L389

![image](https://user-images.githubusercontent.com/2180432/202364429-2cb6c253-192d-422d-a4e8-15d8254927c8.png)

**Reply by phuocthanhdo** [reply]
Or we can completely remove the frame_count from the `FrameTimeDiagnostics`, I think the diagnostics is working fine without it. Current version (0.9), the `FrameTimeDiagnostics` always return frame_count = 0.

**Reply by alice-i-cecile** [reply]
Yeah, I'd be happy to see that. Make a PR?

**Reply by mockersf** [reply]
> the FrameTimeDiagnostics always return frame_count = 0.

No? It works for me and has the correct frame count

**Reply by phuocthanhdo** [reply]
> > the FrameTimeDiagnostics always return frame_count = 0.
> 
> No? It works for me and has the correct frame count

If you include DefaultPlugins in the app, with `default_features`, yes. But as hymm point out, the `update_frame_count` is running from RenderPlugin, so for server side (no render) it doesn't work.

**Reply by phuocthanhdo** [reply]
> Yeah, I'd be happy to see that. Make a PR?

Yeah. I will open a PR

**Comment by phuocthanhdo** [comment]
Just open #6676 for moving `update_frame_count` system

And I found a bit more detail about the source of change:

https://github.com/bevyengine/bevy/pull/5409#issuecomment-1244461420
